### PR TITLE
i#2868 test failures: partial fix mix-native

### PR DESCRIPTION
**Pull request by hgreving2304**
Partial fix for native -> non-native re-takeover. Fix look-up for _dl_fixup by searching for first CALL instruction. IAT style dll call re-takeover is tested in common.nativeexec. Minor logging fixes.

Issue: #2868

### CODE REVIEW

File: core/arch/instr.h [hunk 1]
```diff
@@ -989,7 +989,7 @@ instr_set_target(instr_t *cti_instr, opnd_t target);
 INSTR_INLINE /* hot internally */
 #endif
     DR_API
-    /** Returns true iff \p instr's operands are up to date. */
+    /** Returns true if \p instr's operands are up to date. */
```
Comment (derekbruening) [inline core/arch/instr.h]: "iff" means "if and only if" -- I think we want to leave it

File: core/arch/instr_shared.c [hunk 2]
```diff
@@ -1299,7 +1299,11 @@ instr_expand(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr)
         !instr_valid(instr))
         return instr;
 
-    DOLOG(5, LOG_ALL, { loginst(dcontext, 4, instr, "instr_expand"); });
+    DOLOG(5, LOG_ALL, {
+        instr_t *log_instr = instr_clone(dcontext, instr);
+        loginst(dcontext, 4, log_instr, "instr_expand");
```
Comment (derekbruening) [inline core/arch/instr_shared.c]: I don't understand this one: how does cloning the instr change anything?  This needs a comment explaining, as it is unusual.

File: core/unix/native_elf.c [hunk 3]
```diff
@@ -125,25 +125,23 @@ find_dl_fixup(dcontext_t *dcontext, app_pc resolver)
 {
 #ifdef X86
     instr_t instr;
-    int max_decodes = 30;
+    int max_decodes = 225;
```
Comment (derekbruening) [inline core/unix/native_elf.c]: const

File: core/unix/native_elf.c [hunk 4]
```diff
@@ -125,25 +125,23 @@ find_dl_fixup(dcontext_t *dcontext, app_pc resolver)
...
         pc = decode(dcontext, pc, &instr);
         if (instr_get_opcode(&instr) == OP_call) {
             opnd_t tgt = instr_get_target(&instr);
             fixup = opnd_get_pc(tgt);
-            LOG(THREAD, 1, LOG_LOADER,
+            LOG(THREAD, LOG_LOADER, 1,
                 "%s: found _dl_fixup call at " PFX ", _dl_fixup is " PFX ":\n",
                 __FUNCTION__, pc, fixup);
             break;
-        } else if (instr_is_cti(&instr)) {
```
Comment (derekbruening) [inline core/unix/native_elf.c]: We should still bail on a `ret` I would think.  Continuing through other stuff seems reasonable here.

File: core/arch/instr.h (hunk 1 above)
Comment (hgreving2304) [reply core/arch/instr.h]: never seen!! ok

File: core/unix/native_elf.c (hunk 4 above)
Comment (hgreving2304) [reply core/unix/native_elf.c]: Ok

Comment (hgreving2304) [reply core/unix/native_elf.c]: But not on any cti (there are some in the code)

File: core/arch/instr_shared.c (hunk 2 above)
Comment (hgreving2304) [reply core/arch/instr_shared.c]: The LOG changes the instr (makes operands valid). I don't think we should do that (LOG shouldn't influence the behavior) => I cloned it to make the LOG have its own instr to print.

Comment (hgreving2304) [reply core/arch/instr_shared.c]: Re-open if not ok

Comment (derekbruening) [reply core/arch/instr_shared.c]: I see, that makes sense, just not obvious, so a comment will help.

Comment (derekbruening) [reply core/arch/instr_shared.c]: Just so it's clear, I've been requesting that a comment be added *to the code*, not just this conversation in the review.

Comment (hgreving2304) [reply core/arch/instr_shared.c]: Done

**Pull request merged**